analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldInitializerOrder.cs

```diff
@@ -36,108 +36,64 @@ public sealed class StaticFieldInitializerOrder : SonarDiagnosticAnalyzer
         private const string DiagnosticId = "S3263";
         private const string MessageFormat = "Move this field's initializer into a static constructor.";
 
-        private static readonly DiagnosticDescriptor Rule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+        private static readonly SyntaxKind[] EnclosingTypes =
+        {
+            SyntaxKind.ClassDeclaration,
+            SyntaxKind.InterfaceDeclaration,
+            SyntaxKind.StructDeclaration,
+            SyntaxKindEx.RecordClassDeclaration,
+            SyntaxKindEx.RecordStructDeclaration,
+        };
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
         protected override void Initialize(SonarAnalysisContext context) =>
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c =>
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
                 {
                     var fieldDeclaration = (FieldDeclarationSyntax)c.Node;
                     if (!fieldDeclaration.Modifiers.Any(SyntaxKind.StaticKeyword))
                     {
                         return;
                     }
-
-                    var variables = fieldDeclaration.Declaration.Variables.Where(v => v.Initializer != null).ToList();
-                    if (variables.Count == 0)
+                    var variables = fieldDeclaration.Declaration.Variables.Where(x => x.Initializer != null).ToArray();
```

Comment:
There is no code coverage for cases when there is actually more than one variable with initializers:
```cs
public static int a = 1, b = 2;
```

Reply:
What is the reasoning behind the move from `ToList` to `ToArray`?
I personally prefer `ToList` over `ToArray`, because
* arrays have covariance behavior for reference types, which makes me uncomfortable using them directly (see e.g https://codeblog.jonskeet.uk/2013/06/22/array-covariance-not-just-ugly-but-slow-too/ or https://stackoverflow.com/a/4318510)
* `ToList` internally populates an array while iterating and growing it in chunks. Once the iteration finishes, the last created array is kept even though, it might not be filled entirely. `ToArray` does basically the same, but needs an additional copy at the end to copy to the right-sized array.

I would suggest using either `ToList` or `ToImmutableArray` because both don't have the "whole" in the type system, which may cause runtime errors.

Reply:
I don't remember any single case where we would update the members of the collection after `ToArray()`. So the covariance doesn't concern me.
The difference is in the signature. Returning List declares that the collection we have is expected to change over its lifetime - that items are added/removed somewhere. If we need this, the List is perfectly fine. If we don't then there's no need to wrap the underlying array in the outer List.
The `ImmutableArray` is an option. I prefer it only where we explicitly need state immutability. Otherwise it's just more noisy way to do the same thing.